src/auto-evo/steps/MigrateSpecies.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace AutoEvo;
+
+using System;
+using System.Linq;
+
+public class MigrateSpecies : IRunStep
+{
+    private Patch patch;
+    private SimulationCache cache;
+    private Random random;
+
+    public MigrateSpecies(Patch patch, SimulationCache cache)
+    {
+        this.patch = patch;
+        this.cache = cache;
+        random = new Random();
+    }
+
+    public int TotalSteps => 1;
+
+    public bool CanRunConcurrently => true;
+
+    public bool RunStep(RunResults results)
+    {
+        var miche = results.MicheByPatch[patch];
+
+        var occupants = miche.GetOccupants().Distinct().ToList();
+
+        if (occupants.Count == 0)
+            return true;
+
+        for (int i = 0; i < Constants.AUTO_EVO_MOVE_ATTEMPTS; i++)
+        {
+            var species = occupants.ToList().Random(random);
```

Comment:
Instead of creating a list and immediately discarding it, it would be better to update our random utils to directly be able to use the `occupants` here.